docs: add note for custom command generator template (#7408)
codeigniter4/CodeIgniter4: user_guide_src/source/cli/cli_generators.rst

All generator commands must use the ``GeneratorTrait`` to fully utilize its methods that are used in code
generation.

*************************************
Accessing a Custom Generator Template
*************************************

By default, all generator templates will be looked up at the ``CodeIgniter\Commands\Generators\Views`` namespace.

> To declare a custom location for your custom generator template, you will need to add it to the ``app/Config/Generators.php``

Reviewer: ```suggestion
To declare a custom location for your custom generator template, you will need to add it to the **app/Config/Generators.php**
```

file. For example, if you have a command ``make:awesome-command`` and your generator template is located within your *app*

> directory ``app/Commands/Generators/Views/awesomecommand.tpl.php``, you would update the config file like so:

Reviewer: ```suggestion
directory **app/Commands/Generators/Views/awesomecommand.tpl.php**, you would update the config file like so:
```

> directory **app/Commands/Generators/Views/awesomecommand.tpl.php**, you would update the config file like so:

Reviewer 2: This is wrong. The order of lookup is:
1) The template defined in `app/Config/Generators.php`;
2) The template in the `CodeIgniter\Commands\Generators\Views` namespace.

https://github.com/codeigniter4/CodeIgniter4/blob/848c59860b40355f0de10c1dd9ad3b4761e39b2c/system/CLI/GeneratorTrait.php#L267-L276

Author: Thanks for point that out.

All generator commands must use the ``GeneratorTrait`` to fully utilize its methods that are used in code
generation.

*************************************************************
Declaring the Location of a Custom Generator Command Template
*************************************************************

> The default order of lookup for generator templates is first, the template defined in the **app/Config/Generators.php** file,
> and if not found, the template will be looked up at the ``CodeIgniter\Commands\Generators\Views`` namespace.

Reviewer 2: ```suggestion
The default order of lookup for generator templates is (1) the template defined in the **app/Config/Generators.php** file,
and (2) if not found, the template found at the ``CodeIgniter\Commands\Generators\Views`` namespace.
```
